src/iperf3test_cli.py

```diff
@@ -0,0 +1,161 @@
+
+
+usage = '''iperf3test <command> [<args>]
+
+The available commands are:
+    pps         Creates a new performance test based on a template.
+    bandwidth   Clones an existing performance test.
+'''
+
+
+class IPerf3TestCli:
+
+    def __init__(self):
+        parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                         description='Tests network performance between two machines.', usage=usage)
+        parser.add_argument("command", help="Subcommand to run")
+        args = parser.parse_args(sys.argv[1:2])
+
+        if not hasattr(self, args.command):
+            print("Unrecognized subcommand", parser.print_help())
+            exit(1)
+
+        getattr(self, args.command)()
+
+    def pps(self):
+        parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                         description='Tests max packet count between two machines.',
+                                         usage='''iperf3test pps <ip1> <ip2>''')
+        parser.add_argument("client", help="The client IP address.")
+        parser.add_argument("server", help="The server IP address.")
+        args = parser.parse_args(sys.argv[2:])
+
+        test = IPerf3Test(args.server, args.client)
+        test.run_packet_count_test()
+
+    def bandwidth(self):
+        parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                         description='Tests max bandwidth between two machines.',
+                                         usage='''iperf3test bandwidth <ip1> <ip2>''')
+        parser.add_argument("client", help="The client IP address.")
+        parser.add_argument("server", help="The server IP address.")
+        args = parser.parse_args(sys.argv[2:])
+
+        test = IPerf3Test(args.server, args.client)
+        test.run_bandwidth_test()
+
+
+class IPerf3Test:
+
+    def __init__(self, server_public_ip, client_public_ip):
+        inventory = load_inventory()
+
+        self.client = find_host_with_public_ip(inventory, client_public_ip)
+        if self.client is None:
+            exit_with_error(f"Could not find client [{client_public_ip} in the inventory]")
+
+        self.server = find_host_with_public_ip(inventory, server_public_ip)
+        if self.server is None:
+            exit_with_error(f"Could not find server [{server_public_ip} in the inventory]")
+
+    def run_bandwidth_test(self):
+        client_args = "--parallel 1 --time 20"
+        server_args = ""
+
+        server_thread = threading.Thread(target=start_server, args=(self.server, server_args))
+        server_thread.start()
+
+        client_thread = threading.Thread(target=start_client, args=(self.client, self.server, client_args))
+        client_thread.start()
+
+    def run_packet_count_test(self):
+        run_duration = 20
+        client_args = f"--parallel 128 --time {run_duration} --set-mss 89"
```

Comment:
I run this with high parallelism, to ensure we get max pps.
128 might seem excessive and initially I have used 10 streams, but running some tests against a c5n instance I found that I'm not able to get consistent pps results.
And running with 128 streams does not seem to have any drawbacks, on smaller instances I have consistently achieved the same results as with 10 streams.